dex-jmh/src/test/scala/com/wavesplatform/dex/model/address/AddressActorStartingBenchmark.scala

```diff
@@ -0,0 +1,177 @@
+                Seq(getOrderAddedEvent(ao), firstExecutedEvent, secondExecutedEvent)
+              }
+            case OrderEvent.PartiallyFilled =>
+              ordersGroup.flatMap { ao =>
+                val firstExecutedEvent  = getOrderExecutedEvent(ao, getCounter(ao, ao.amount / 2))
+                val secondExecutedEvent = getOrderExecutedEvent(firstExecutedEvent.submittedRemaining, getCounter(ao, ao.amount / 3))
+                Seq(getOrderAddedEvent(ao), firstExecutedEvent, secondExecutedEvent)
+              }
+            case OrderEvent.PartiallyFilledAndThenCancelled =>
+              ordersGroup.flatMap { ao =>
+                val firstExecutedEvent = getOrderExecutedEvent(ao, getCounter(ao, ao.amount / 2))
+                val cancelEvent        = getOrderCancelledEvent(firstExecutedEvent.submittedRemaining)
+                Seq(getOrderAddedEvent(ao), firstExecutedEvent, cancelEvent)
+              }
+          }
+        (result :+ eventsGroup) -> newRemaining
+      }
+
+    implicit val efc: ErrorFormatterContext = ErrorFormatterContext.from(_ => 8)
+
+    def run(): Unit = {
+
+      val system: ActorSystem = ActorSystem("addressActorBenchmark")
+
+      val addressActor: ActorRef =
+        system.actorOf(
+          AddressActor.props(
+            owner = owner,
+            time = new TestTime(),
+            orderDB = EmptyOrderDB,
+            validate = (_, _) => Future.successful(().asRight),
+            store = event => Future.successful { Some(QueueEventWithMeta(0L, 0L, event)) },
+            started = true,
+            spendableBalancesActor = ActorRef.noSender,
+            settings = AddressActor.Settings.default
+          )
+        )
+
+      for {
+        events <- Random.shuffle(eventsGroups)
+        event  <- events
+      } addressActor ! event
+
+      Await.result(addressActor.ask(Query.GetReservedBalance)(3.minutes), 3.minutes)
+      Await.result(system.terminate(), 3.seconds)
+    }
+
+    def ordersGen(orderNumber: Int): Gen[List[AcceptedOrder]] =
+      for {
+        orderSides <- Gen.listOfN(orderNumber, orderSideGen)
+        orders <- Gen.sequence {
+          orderSides.map { side =>
+            val orderGen = if (side == OrderType.SELL) askGen else bidGen
+            Gen.oneOf(limitOrderGen(orderGen), marketOrderGen(orderGen))
+          }
+        }
+      } yield orders.asScala.toList
+
+    def getOrderAddedEvent(ao: AcceptedOrder): Events.OrderAdded =
+      Events.OrderAdded(ao, Events.OrderAddedReason.RequestExecuted, System.currentTimeMillis())
+
+    def getOrderCancelledEvent(ao: AcceptedOrder): Events.OrderCanceled =
+      Events.OrderCanceled(ao, Events.OrderCanceledReason.RequestExecuted, System.currentTimeMillis())
+
+    def getCounter(ao: AcceptedOrder, amount: Long): LimitOrder = {
+      LimitOrder(
+        ao.order
+          .updateType(ao.order.orderType.opposite)
+          .updateSender(clientGen.sample.get)
```

Comment:
Or create this once above. It doesn't matter, who is the client of the counter order (but it is better to provide a different client)